### PR TITLE
Increase the upper limit on group name length.

### PR DESCRIPTION
**Pull request by anhu**
While all the standardized groups would fit within the old limit,
with the addition of providers, some might want to experiment with
new and unstandardized groups. As such, their names might not fit
within the old limit.

### CODE REVIEW

**Comment by anhu** [comment]
Trivial.

**Comment by anhu** [comment]
Thanks!!

File: ssl/t1_lib.c [hunk 1]
```diff
@@ -705,7 +705,7 @@ static int gid_cb(const char *elem, int len, void *arg)
     gid_cb_st *garg = arg;
     size_t i;
     uint16_t gid = 0;
-    char etmp[20];
+    char etmp[64];
```
Comment (t8m) [inline ssl/t1_lib.c:708]: Should the constant be a # define somewhere in some include file at least?

Comment (anhu) [reply ssl/t1_lib.c:708]: There are constants like this everywhere in the code.  If it were to be a `#define`  , I would imagine that to be part of a wider code cleanup effort.  

However, I am willing to do it.  Can you suggest a name and location?  I'm thinking `ssl_local.h`.

**Comment by anhu** [comment]
Tests detected a memory leak, but I don't think my change could have caused a memory leak.

**Comment by mattcaswell** [comment]
> Tests detected a memory leak, but I don't think my change could have caused a memory leak.

No. This is a known failure in the CIs at the moment.

File: ssl/t1_lib.c (hunk 1 above)
Comment (anhu) [reply ssl/t1_lib.c]: I defined it right after `MAX_GROUPLIST`

**Review by mattcaswell — APPROVED** [review]
I'm still ok with this.

File: ssl/t1_lib.c [hunk 2]
```diff
@@ -693,7 +693,7 @@ int tls1_set_groups(uint16_t **pext, size_t *pextlen,
 
 /* TODO(3.0): An arbitrary amount for now. Take another look at this */
 # define MAX_GROUPLIST   40
-
+# define GROUP_NAME_BUFFER_LENGTH 64
```
Comment (t8m) [inline ssl/t1_lib.c:696]: For cosmetic reasons I'd prefer keeping empty line after the defines. But OK anyway.

**Review by t8m — APPROVED** [review]

**Review by paulidale — APPROVED** [review]
Agreed, trivial.

**Comment by t8m** [comment]
Merged to master. Thank you for the contribution.

**Pull request closed without merging**